File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/fight/arena/GameRolePlayArenaRegisterMessage.scala





















// Generated on 05/03/2013 10:34:43
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameRolePlayArenaRegisterMessage extends MessageDeserializer {
  val messageId = 6280

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameRolePlayArenaRegisterMessage.messageId

    val battleMode: Int

    def serialize(buf: Buffer) {
      buf.writeInt(battleMode)
    }

  }

  protected def create(battleMode: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val battleMode = buf.readInt()

    create(battleMode)
  }

}

object GameRolePlayArenaRegisterMessage extends GameRolePlayArenaRegisterMessage {
  case class MessageImpl(battleMode: Int) extends Message
  type Target = MessageImpl

  override def create(battleMode: Int) = MessageImpl(battleMode)

  def apply(battleMode: Int) = create(battleMode)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameRolePlayArenaRegisterMessage.java


// Generated on 05/08/2013 19:37:46
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameRolePlayArenaRegisterMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6280;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int battleMode;
    
    public GameRolePlayArenaRegisterMessage() { }
    
    public GameRolePlayArenaRegisterMessage(int battleMode) {
        this.battleMode = battleMode;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(battleMode);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        battleMode = buf.readInt();
        if (battleMode < 0)
            throw new RuntimeException("Forbidden value on battleMode = " + battleMode + ", it doesn't respect the following condition : battleMode < 0");
    }
    
}
